app/assets/javascripts/application.js

```diff
@@ -37,6 +37,9 @@ var PAGE_RENDERED = false;
 var OPEN_MODAL = '';
 var CLUSTER_TYPE = '3d';
 var UNSAFE_CHARACTERS = /[\;\/\?\:\@\=\&\'\"\<\>\#\%\{\}\|\\\^\~\[\]\`]/g;
+var MAX_GENE_SEARCH = 20;
+var GENE_SEARCH_LIMIT_MSG = 'For performance reasons, gene search is limited to ' + MAX_GENE_SEARCH + ' genes.  ' +
```

Comment:
Having two copies of these lines -- one here and in one in `app/models/gene.rb` -- will require vigilance to keep in sync.  Ideas to mitigate that:

* Assign these two variables in a `<script>` tag atop the Rails view template [application.html.erb](https://github.com/broadinstitute/single_cell_portal_core/blob/035a8250d9d8354ddd21115fd30532bba0f92ead/app/views/layouts/application.html.erb#L9), using a value defined in server code.  E.g. `window.MAX_GENE_SEARCH = <%= @my_object.MAX_GENE_SEARCH %>`. Assign them in `app/models/gene.rb` by referencing the same server code.

* Add a comment here and in `app/models/gene.rb` to change the other place when updating either value.

(If they're intended to be independent, then having names that suggest they're different -- e.g. incorporating `GLOBAL` and `STUDY` into the variable names -- would help maintainers.)

Reply:
They're not intended to be independent, I just couldn't think of an easy way to use the value from Ruby in `application.js`.  But your idea of using `window` in `application.html.erb` will solve that nicely.  I'll make that change.